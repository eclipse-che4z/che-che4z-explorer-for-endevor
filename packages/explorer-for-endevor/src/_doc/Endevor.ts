/*
 * Â© 2023 Broadcom Inc and/or its subsidiaries; All rights reserved
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Broadcom, Inc. - initial API and implementation
 */

import {
  ChangeControlValue,
  StageNumber,
  Value,
} from '@local/endevor/_doc/Endevor';

export type EnvironmentStageSearchLocation = Readonly<{
  environment: Value;
  stageNumber: StageNumber;
}>;

// we do not use wildcards as a values.
// every value is uppercased by default, except the configuration.
export type ElementSearchLocation = Partial<
  Readonly<{
    configuration: Value;
  }>
> &
  Partial<EnvironmentStageSearchLocation> &
  Partial<
    Readonly<{
      system: Value;
      subsystem: Value;
      type: Value;
      element: Value;
    }>
  > &
  Partial<ChangeControlValue>;

export type SearchLocation = Omit<ElementSearchLocation, 'configuration'>;

// any subsystem in any system in the search environment & stage number
export type SubsystemMapPathId = string;
// code routes up the Endevor map
type Routes = ReadonlyArray<SubsystemMapPathId>;

export type EndevorMap = Readonly<{
  [endevorMapNode: SubsystemMapPathId]: Routes;
}>;
