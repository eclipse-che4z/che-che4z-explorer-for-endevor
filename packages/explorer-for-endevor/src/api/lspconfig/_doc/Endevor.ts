/*
 * Â© 2023 Broadcom Inc and/or its subsidiaries; All rights reserved
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Broadcom, Inc. - initial API and implementation
 */

import {
  ResponseStatus,
  SuccessEndevorResponse as SuccessResponse,
} from '@local/endevor/_doc/Endevor';
import { ExternalConfigurationResponse } from './Configuration';

export const enum ErrorResponseType {
  GENERIC_ERROR = 'GENERIC_ERROR',
  IMPORT_ERROR = 'IMPORT_ERROR',
}
export type ErrorType = ErrorResponseType.GENERIC_ERROR;
export type ErrorResponse<E extends ErrorResponseType | undefined = undefined> =
  Readonly<{
    status: ResponseStatus.ERROR;
    type: (E extends undefined ? never : E) | ErrorType;
    details: Readonly<{
      messages: ReadonlyArray<string>;
    }>;
  }>;

export type Response<
  E extends ErrorResponseType | undefined = undefined,
  R = undefined
> = SuccessResponse<R> | ErrorResponse<E>;

export type ConfigurationResponse = Response<
  ErrorResponseType.IMPORT_ERROR,
  ExternalConfigurationResponse
>;
