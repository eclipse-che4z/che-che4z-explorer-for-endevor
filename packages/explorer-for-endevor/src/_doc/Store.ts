/*
 * Â© 2022 Broadcom Inc and/or its subsidiaries; All rights reserved
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Broadcom, Inc. - initial API and implementation
 */

import { BaseCredential } from '@local/endevor/_doc/Credential';
import { Element } from '@local/endevor/_doc/Endevor';
import { ElementLocationName, EndevorServiceName } from './settings';

export type CachedElement = {
  element: Element;
  lastRefreshTimestamp: number;
};

export type CachedElements = Readonly<{
  [id: string]: CachedElement;
}>;

export type EndevorCacheItem = Readonly<{
  searchLocation: ElementLocationName;
  elements: CachedElements;
}>;

export type StateItem = {
  serviceName: EndevorServiceName;
  credential?: BaseCredential;
  cachedElements: ReadonlyArray<EndevorCacheItem>;
};

export type State = ReadonlyArray<StateItem>;
