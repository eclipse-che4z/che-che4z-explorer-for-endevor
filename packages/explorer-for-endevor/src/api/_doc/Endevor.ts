/*
 * Â© 2023 Broadcom Inc and/or its subsidiaries; All rights reserved
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Broadcom, Inc. - initial API and implementation
 */

import { Credential } from '@local/endevor/_doc/Credential';
import {
  ChangeControlValue,
  ServiceLocation,
  StageNumber,
  Value,
} from '@local/endevor/_doc/Endevor';

// TODO move to the endevor package to be reused on the API level too
// TODO rejectUnauthorized may be put in dependency on the protocol which is used
export type EndevorUnauthorizedService = Readonly<{
  location: ServiceLocation;
  rejectUnauthorized: boolean;
}>;

// TODO move to the endevor package to be reused on the API level too
export type EndevorAuthorizedService = EndevorUnauthorizedService &
  Readonly<{
    configuration: string;
    credential: Credential;
  }>;

export type EnvironmentStageSearchLocation = Readonly<{
  environment: Value;
  stageNumber: StageNumber;
}>;

// we do not use wildcards as a values.
// every value is uppercased by default, except the configuration.
export type ElementSearchLocation = Partial<
  Readonly<{
    configuration: Value;
  }>
> &
  Readonly<EnvironmentStageSearchLocation> &
  Partial<
    Readonly<{
      system: Value;
      subsystem: Value;
      type: Value;
      element: Value;
    }>
  > &
  Partial<ChangeControlValue>;

export type SearchLocation = Omit<ElementSearchLocation, 'configuration'>;

// any environment & stage number
export type EnvironmentStageMapPathId = string;
// any system in the search environment & stage number
export type SystemMapPathId = string;
// any subsystem in any system in the search environment & stage number
export type SubsystemMapPathId = string;
// any type in any system in the search environment & stage number
export type TypeMapPathId = string;

// code routes up the Endevor map
type Routes = ReadonlyArray<SubsystemMapPathId>;

export type EndevorMap = Readonly<{
  [endevorMapNode: SubsystemMapPathId]: Routes;
}>;
