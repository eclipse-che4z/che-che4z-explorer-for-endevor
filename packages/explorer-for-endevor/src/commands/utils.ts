/*
 * Â© 2023 Broadcom Inc and/or its subsidiaries; All rights reserved
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Broadcom, Inc. - initial API and implementation
 */

import { toServiceLocationCompositeKey } from '../store/utils';
import { ElementNode } from '../tree/_doc/ElementTree';
import { Uri } from 'vscode';
import { showFileContent } from '@local/vscode-wrapper/window';

type GroupedElementNodes = {
  [searchLocationId: string]: ReadonlyArray<ElementNode>;
};
export const groupBySearchLocationId = (
  elementNodes: ReadonlyArray<ElementNode>
): Readonly<GroupedElementNodes> => {
  return elementNodes.reduce((acc: GroupedElementNodes, currentNode) => {
    const serviceLocationId = toServiceLocationCompositeKey(
      currentNode.serviceId
    )(currentNode.searchLocationId);
    acc[serviceLocationId] = [...(acc[serviceLocationId] || []), currentNode];
    return acc;
  }, {});
};

export const showElementToEdit = async (
  fileUri: Uri
): Promise<void | Error> => {
  try {
    await showFileContent(fileUri);
  } catch (e) {
    return new Error(
      `Unable to open the file ${fileUri.fsPath} because of error ${e.message}`
    );
  }
};
