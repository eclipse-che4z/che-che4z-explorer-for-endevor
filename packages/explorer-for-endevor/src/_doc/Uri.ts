/*
 * Â© 2023 Broadcom Inc and/or its subsidiaries; All rights reserved
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Broadcom, Inc. - initial API and implementation
 */

// TODO: rewrite with io-ts types
import {
  ChangeControlValue,
  Element,
  ElementMapPath,
  Service,
  SubSystemMapPath,
  Value,
} from '@local/endevor/_doc/Endevor';
import { EndevorId } from '../store/_doc/v2/Store';

export const enum Schemas {
  TREE_ELEMENT = 'e4eElement',
  ELEMENT_LISTING = 'e4eListing',
  ELEMENT_HISTORY = 'e4eHistory',
  ELEMENT_CHANGE_LVL = 'e4eChangeLvl',
  FILE = 'file',
  READ_ONLY_FILE = 'e4eReadonlyFile',
  READ_ONLY_CACHED_ELEMENT = 'e4eReadOnlyCachedElement',
  READ_ONLY_REPORT = 'e4eReadonlyReport',
  READ_ONLY_GENERIC_REPORT = 'e4eReadonlyGenericReport',
}

export const enum QueryTypes {
  EDITED_ELEMENT = 'edited-element',
  COMPARED_ELEMENT = 'compared-element',
}

export const enum Extensions {
  TREE_ELEMENT = 'prnt',
  ELEMENT_LISTING = 'lst',
  ELEMENT_HISTORY = 'hist',
  ELEMENT_CHANGE_LVL = 'chng',
  ACTION_REPORT = 'rep',
}

export type EditedElementUriQuery = Readonly<{
  element: Element;
  fingerprint: string;
  searchContext: {
    initialSearchLocation: SubSystemMapPath;
    // TODO: remove from the URI, not related to the element itself
    serviceId: EndevorId;
    // TODO: remove from the URI, not related to the element itself
    searchLocationId: EndevorId;
  };
}>;

export type ComparedElementUriQuery = Readonly<{
  element: Element;
  fingerprint: string;
  remoteVersionTempFilePath: string;
  initialSearchContext: {
    initialSearchLocation: SubSystemMapPath;
    // TODO: remove from the URI, not related to the element itself
    serviceId: EndevorId;
    // TODO: remove from the URI, not related to the element itself
    searchLocationId: EndevorId;
  };
  uploadChangeControlValue: ChangeControlValue;
  uploadTargetLocation: ElementMapPath;
}>;

export type BasicElementUriQuery = Readonly<{
  element: Element;
  serviceId: EndevorId;
  searchLocationId: EndevorId;
}>;

export type ElementListingUriQuery = BasicElementUriQuery;

export type ElementHistoryUriQuery = BasicElementUriQuery;

export type ElementChangeUriQuery = ElementHistoryUriQuery &
  Readonly<{
    vvll: string;
  }>;

export type FragmentType = {
  fragment: string;
};

export type ActionReportUriQuery = Readonly<{
  service: Service;
  configuration: Value;
  reportId: Value;
  objectName: Value;
  serviceId: EndevorId;
  searchLocationId: EndevorId;
  ccid: Value;
}>;
