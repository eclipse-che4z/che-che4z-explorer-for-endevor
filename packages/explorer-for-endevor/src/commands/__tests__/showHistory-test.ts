/*
 * Â© 2023 Broadcom Inc and/or its subsidiaries; All rights reserved
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Broadcom, Inc. - initial API and implementation
 */

import assert = require('assert');
import * as sinon from 'sinon';
import { CachedElement } from '../../store/_doc/v2/Store';
import { parseHistory } from '../../tree/endevor';
import * as elementHistoryUri from '../../uri/elementHistoryUri';
import { isError } from '../../utils';
import { getChangeLevelContent } from '../../view/changeLvlContentProvider';
import { Element } from '@local/endevor/_doc/Endevor';
import { Id, Source } from '../../store/storage/_doc/Storage';
import { UNIQUE_ELEMENT_FRAGMENT } from '../../constants';
import { Uri } from 'vscode';
import { ElementChangeUriQuery, FragmentType } from '../../_doc/Uri';

jest.mock(
  'vscode',
  () => ({
    window: {
      createTextEditorDecorationType: jest.fn(),
    },
  }),
  { virtual: true }
);
jest.mock(
  '../../globals',
  () => ({
    logger: {
      trace: jest.fn(),
      info: jest.fn(),
      warn: jest.fn(),
      error: jest.fn(),
    },
    reporter: {
      sendTelemetryEvent: jest.fn(),
    },
  }),
  { virtual: true }
);

const historyTextValid = `1CA Endevor SCM Version 18.1.00        Copyright (C) 1986-2019 Broadcom. All Rig
PROBLEM                                               CA Endevor SCM           
PRINT                                  HISTORY                                 
ELEMENT: ELMCAST1                                                              
                                                                               
*******************************************************************************
*******************************************************************************
**                                                                           **
** ELEMENT HISTORY                                           28FEB23  04:14  **
**                                                                           **
**    ENVIRONMENT:   ENV1       SYSTEM:    QAPKG     SUBSYSTEM:  SBSQAM2     **
**    TYPE:          ASM        STAGE ID:  A                                 **
**    ELEMENT:       ELMCAST1                                                **
**                                                                           **
**    SIGNED OUT TO: MILANKO                         DELTA TYPE: REVERSE     **
**                                                                           **
*******************************************************************************
*******************************************************************************
                                                                               
-------------------------- SOURCE LEVEL INFORMATION ---------------------------
                                                                               
  VVLL SYNC USER     DATE    TIME     STMTS CCID         COMMENT                                   
  ---- ---- -------- ------- ----- -------- ------------ ----------------------------------------  
  0100      MILANKO  31JAN23 04:28        1                                                        
  0101      MILANKO  28FEB23 04:13        2 TEST         COMMENT                                   
  0102      JANMI04  29FEB23 04:14        3 HELE         HALUZ                                     
  0103      USERXYZ  30FEB23 04:14        2 CCID                                                   
  0104      MILANKO  28FEB23 04:14        3              JUST COMMENT                              
GENERATED   MILANKO  28FEB23 04:14                                                                 
                                                                                                   
 +0100      ELEMENT                                                                         
%+0104      LINE 4                                                                          
%+0104      LINE 5                                                                          
%+0101-0103 LINE 2                                                                          
%+0102-0104 LINE 3                                                                          
`;

const historyTextInvalid = `1CA Endevor SCM Version 18.1.00        Copyright (C) 1986-2019 Broadcom. All Rig
PROBLEM                                               CA Endevor SCM           
PRINT                                  HISTORY                                 
ELEMENT: ELMCAST1                                                              
                                                                               
*******************************************************************************
*******************************************************************************
**                                                                           **
** ELEMENT HISTORY                                           28FEB23  04:14  **
**                                                                           **
**    ENVIRONMENT:   ENV1       SYSTEM:    QAPKG     SUBSYSTEM:  SBSQAM2     **
**    TYPE:          ASM        STAGE ID:  A                                 **
**    ELEMENT:       ELMCAST1                                                **
**                                                                           **
**    SIGNED OUT TO: MILANKO                         DELTA TYPE: REVERSE     **
**                                                                           **
*******************************************************************************
*******************************************************************************
                                                                               
-------------------------- SOURCE LEVEL INFORMATION ---------------------------
                                                                               
  VVLL SYNC USER     DATE    TIME     STMTS CCID            COMMENT                                   
  ---- ---- -------- ------- ----- -------- ------------    ----------------------------------------  
  0100      MILANKO  31JAN23 04:28        1                                                        
  0101      MILANKO  28FEB23 04:13        2 TEST            COMMENT                                   
  0102      JOZKO    29FEB23 04:14        3 HELE            HALUZ                                     
  0103      USERXYZ  30FEB23 04:14        2 CCID                                                   
  0104      MILANKO  28FEB23 04:14        3                 JUST COMMENT                              
GENERATED   MILANKO  28FEB23 04:14                                                                 
                                                                                                   
 +0100      ELEMENT                                                                         
%+0104      LINE 4                                                                          
%+0104      LINE 5                                                                          
%+0101-0103 LINE 2                                                                          
%+0102-0104 LINE 3                                                                          
`;

describe('showHistory', () => {
  const uri = {
    scheme: 'hist',
    authority: '',
    path: '',
    query: '',
    fragment: '',
    fsPath: '',
    with: jest.fn(),
    toJSON: jest.fn(),
  };
  describe('parseHistory', () => {
    it('should properly parse the history content', () => {
      const historyData = parseHistory(historyTextValid, uri);
      if (isError(historyData)) {
        const error = historyData;
        assert.fail(`Could not parse history because of: ${error.message}`);
      }
      expect(historyData.changeLevels.length).toBe(5);
      expect(historyData.historyLines.length).toBe(5);
      expect(historyData.changeLevels[1]).toStrictEqual({
        uri,
        vvll: '0101',
        user: 'MILANKO',
        date: '28FEB23',
        time: '04:13',
        ccid: 'TEST',
        comment: 'COMMENT',
        lineNums: [],
      });
      expect(historyData.changeLevels[3]).toStrictEqual({
        uri,
        vvll: '0103',
        user: 'USERXYZ',
        date: '30FEB23',
        time: '04:14',
        ccid: 'CCID',
        comment: '',
        lineNums: [],
      });
      expect(historyData.changeLevels[4]).toStrictEqual({
        uri,
        vvll: '0104',
        user: 'MILANKO',
        date: '28FEB23',
        time: '04:14',
        ccid: '',
        comment: 'JUST COMMENT',
        lineNums: [],
      });
      // check history lines
      expect(historyData.historyLines[0]).toStrictEqual({
        changed: false,
        line: 30,
        removedVersion: '',
        addedVersion: '0100',
        lineLength: 92,
      });
      expect(historyData.historyLines[2]).toStrictEqual({
        changed: true,
        line: 32,
        removedVersion: '',
        addedVersion: '0104',
        lineLength: 92,
      });
      expect(historyData.historyLines[3]).toStrictEqual({
        changed: true,
        line: 33,
        removedVersion: '0103',
        addedVersion: '0101',
        lineLength: 92,
      });
    });

    it('should return error if history change levels not in correct format', () => {
      const historyData = parseHistory(historyTextInvalid, uri);
      expect(isError(historyData)).toBe(true);
    });
  });

  describe('getChangeLevelContent', () => {
    afterEach(async () => {
      // Sinon has some issues with cleaning up the environment after itself, so we have to do it
      // TODO: take a look into Fake API instead of Stub
      sinon.restore();
    });

    const dispatch = jest.fn();
    const serviceId: Id = { name: 'serviceName', source: Source.INTERNAL };
    const searchLocationId: Id = {
      name: 'searchLocationName',
      source: Source.INTERNAL,
    };
    const element: Element = {
      environment: 'ENV',
      system: 'SYS',
      subSystem: 'SUBSYS',
      stageNumber: '1',
      type: 'TYP',
      name: 'ELM',
      id: 'ELM',
      noSource: false,
      extension: 'ext',
      lastActionCcid: 'LAST-CCID',
    };
    const getElement = () => () => (): CachedElement | undefined => {
      return {
        element,
        elementIsUpTheMap: false,
        lastRefreshTimestamp: Date.now(),
        historyData: {
          changeLevels: [
            {
              uri,
              vvll: '0100',
              user: 'MELDO',
              date: '010101',
              time: 'HH:MM',
              ccid: 'CCID',
              comment: 'COMMENT',
            },
            {
              uri,
              vvll: '0101',
              user: 'MELDO',
              date: '010101',
              time: 'HH:MM',
              ccid: 'CCID',
              comment: 'COMMENT',
            },
            {
              uri,
              vvll: '0102',
              user: 'MELDO',
              date: '010101',
              time: 'HH:MM',
              ccid: 'CCID',
              comment: 'COMMENT',
            },
            {
              uri,
              vvll: '0103',
              user: 'MELDO',
              date: '010101',
              time: 'HH:MM',
              ccid: 'CCID',
              comment: 'COMMENT',
            },
            {
              uri,
              vvll: '0104',
              user: 'MELDO',
              date: '010101',
              time: 'HH:MM',
              ccid: 'CCID',
              comment: 'COMMENT',
            },
          ],
          historyLines: [
            {
              addedVersion: '0100',
              line: 30,
              removedVersion: '',
              lineLength: 92,
            },
            {
              addedVersion: '0104',
              line: 31,
              removedVersion: '',
              lineLength: 92,
            },
            {
              addedVersion: '0104',
              line: 32,
              removedVersion: '',
              lineLength: 92,
            },
            {
              addedVersion: '0101',
              line: 33,
              removedVersion: '0103',
              lineLength: 92,
            },
            {
              addedVersion: '0102',
              line: 34,
              removedVersion: '0104',
              lineLength: 92,
            },
          ],
        },
      };
    };

    const mockGetUri =
      (uriArg: Uri) =>
      (mockResult: (ElementChangeUriQuery & FragmentType) | Error) => {
        return sinon
          .stub(elementHistoryUri, 'fromElementChangeUri')
          .withArgs(uriArg)
          .returns(mockResult);
      };

    it('creates correct content for level 0100', async () => {
      mockGetUri(uri)({
        serviceId,
        searchLocationId,
        element,
        vvll: '0100',
        fragment: UNIQUE_ELEMENT_FRAGMENT,
      });
      const changeLevelContent = await getChangeLevelContent(
        dispatch,
        getElement,
        uri,
        historyTextValid
      );
      expect(changeLevelContent).toMatchSnapshot();
    });

    it('creates correct content for level 0101', async () => {
      mockGetUri(uri)({
        serviceId,
        searchLocationId,
        element,
        vvll: '0101',
        fragment: UNIQUE_ELEMENT_FRAGMENT,
      });
      const changeLevelContent = await getChangeLevelContent(
        dispatch,
        getElement,
        uri,
        historyTextValid
      );
      expect(changeLevelContent).toMatchSnapshot();
    });

    it('creates correct content for level 0103', async () => {
      mockGetUri(uri)({
        serviceId,
        searchLocationId,
        element,
        vvll: '0103',
        fragment: UNIQUE_ELEMENT_FRAGMENT,
      });
      const changeLevelContent = await getChangeLevelContent(
        dispatch,
        getElement,
        uri,
        historyTextValid
      );
      expect(changeLevelContent).toMatchSnapshot();
    });

    it('creates correct content for level 0104', async () => {
      mockGetUri(uri)({
        serviceId,
        searchLocationId,
        element,
        vvll: '0104',
        fragment: UNIQUE_ELEMENT_FRAGMENT,
      });
      const changeLevelContent = await getChangeLevelContent(
        dispatch,
        getElement,
        uri,
        historyTextValid
      );
      expect(changeLevelContent).toMatchSnapshot();
    });
  });
});
