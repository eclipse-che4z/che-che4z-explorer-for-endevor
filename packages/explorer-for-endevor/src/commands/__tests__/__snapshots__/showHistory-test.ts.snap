// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showHistory getChangeLevelContent creates correct content for level 0100 1`] = `
"ELEMENT                                                                         
"
`;

exports[`showHistory getChangeLevelContent creates correct content for level 0101 1`] = `
"ELEMENT                                                                         
LINE 2                                                                          
"
`;

exports[`showHistory getChangeLevelContent creates correct content for level 0103 1`] = `
"ELEMENT                                                                         
LINE 2                                                                          
LINE 3                                                                          
"
`;

exports[`showHistory getChangeLevelContent creates correct content for level 0104 1`] = `
"ELEMENT                                                                         
LINE 4                                                                          
LINE 5                                                                          
LINE 3                                                                          
"
`;
